File: server/internal/dto/company.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

// CreateCompanyRequest is the payload for creating a new company.
type CreateCompanyRequest struct {
	Name        string `json:"name"        binding:"required"`
	Description string `json:"description"`
	Logo        string `json:"logo"`
	Website     string `json:"website"`
	Industry    string `json:"industry"`
}

// UpdateCompanyRequest allows updating mutable company fields.
type UpdateCompanyRequest struct {
	Name        string `json:"name"`
	Description string `json:"description"`
	Logo        string `json:"logo"`
	Website     string `json:"website"`
	Industry    string `json:"industry"`
}

// CompanyResponse is the public-facing company object.
// Uses camelCase to match Flutter's Company.fromJson.
type CompanyResponse struct {
	ID          uuid.UUID `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Logo        string    `json:"logo"`
	Website     string    `json:"website"`
	Industry    string    `json:"industry"`
	OwnerID     uuid.UUID `json:"ownerId"`
	Status      string    `json:"status"`
	MemberCount int64     `json:"memberCount"`
	FlowCount   int64     `json:"flowCount"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

// AddMemberRequest adds a user to a company with an optional role.
type AddMemberRequest struct {
	UserID uuid.UUID `json:"userId" binding:"required"`
	RoleID uuid.UUID `json:"roleId" binding:"required"`
}

File: api_cli/lib/models/company.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'company.freezed.dart';
part 'company.g.dart';

@freezed
class Company with _$Company {
  const factory Company({
    required String id,
    required String name,
    String? logo,
    String? description,
    String? website,
    String? industry,
    String? ownerId,
    @Default('active') String status,
    @Default(0) int memberCount,
    @Default(0) int flowCount,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) = _Company;

  factory Company.fromJson(Map<String, dynamic> json) => _$CompanyFromJson(json);
}
